pyccel/parser/semantic.py

```diff
@@ -356,6 +368,34 @@ def check_for_variable(self, name):
         else:
             return self.scope.find(name, 'variables')
 
+    def check_for_attribute(self, name):
+        """
+        Search for an Attribute with the given name in the current scope.
+
+        Search for an Attribute with the given name in the current scope,
+        defined by the local and global Python scopes. Return None if not found.
+
+        Parameters
+        ----------
+        name : DottedName
+            The object describing the attribute.
+
+        Returns
+        -------
+        Variable
+            Returns the attribute if found or None.
+        """
+
+        if self.current_class:
+            attributes_list = list(self._current_class.attributes)
+            for key, value in enumerate(attributes_list):
+                if value.name == name.name[-1]:
+                    attributes_list[key] = value
+                    self._current_class.attributes = tuple(attributes_list)
```

Comment:
What are you doing here?
The attributes exist, e.g. they are `('a', 'b', 'c')` and we are looking for `'c'`
You get a list: `['a', 'b', 'c']`, find `'c'`, insert the `'c'` from the list into the same location in the list (list unchanged) then update the attributes with their original values (attributes unchanged)

Reply:
Fixed.